audio-worklet/design-pattern/awn-ring-buffer/RingBufferWorkletProcessorKernel.cc

```diff
@@ -0,0 +1,46 @@
+#include "emscripten/bind.h"
+
+using namespace emscripten;
+
+// A multi-channel processing kernel. To handle multiple inputs or outputs,
+// simply use multiple instances of this kernel. The design assumes:
+//   1. The kernel size (processing frame size) is static after construction.
+//   2. The channel count can be changed dynamically.
+//   3. The given channel count must be smaller than the size of the array
+//      being passed in.
+class RingBufferWorkletProcessorKernel {
+ public:
+  RingBufferWorkletProcessorKernel(unsigned kernel_buffer_size)
```

Comment:
Naming is a bit confusing.  `kernel_buffer_size` is basically the size of the FIFO?

Reply:
Although the class name contains "RingBuffer" in it, this class does not implement the ring buffer. Yes, technically the `kernel_buffer_size` is equal to the size of RingBuffer - that's because the user function (the kernel) requires a certain buffer size, and the RingBuffer is there to accommodate such buffer size within AudioWorkletProcessor.

Perhaps taking "RingBuffer" out of the class name would be better?

Reply:
Then it becomes WorkletProcessorKernel, which is kind of generic.  Maybe WorkletProcessorKernelWithRingBuffer (or WithFIFO)?

I'll leave the final choice to you.